src/diagnostics.rs

```diff
@@ -121,13 +121,13 @@ pub fn report_error<'tcx, 'mir>(
 /// Report an error or note (depending on the `error` argument) at the current frame's current statement.
 /// Also emits a full stacktrace of the interpreter stack.
 fn report_msg<'tcx, 'mir>(
-    ecx: &InterpCx<'mir, 'tcx, Evaluator<'tcx>>,
+    ecx: &InterpCx<'mir, 'tcx, Evaluator<'mir, 'tcx>>,
     title: &str,
     span_msg: String,
     mut helps: Vec<String>,
     error: bool,
 ) -> Option<i64> {
-    let span = if let Some(frame) = ecx.stack().last() {
+    let span = if let Some(frame) = ecx.active_thread_stack().last() {
```

Comment:
Are these changes necessary? Seems like the only code should work just as well?

Reply:
Now yes... 🤦 When you asked me to make the `InterpCx::stack_mut` method `pub(crate)` because the `stack` field was originally `pub(crate)`, I did not realize that `InterpCx::stack` was a preexisting public method and I made [both](https://github.com/rust-lang/rust/pull/70598/files#diff-470c35c1f39f9d3f8817435a63c73ad2L350-R356)  methods (`stack` and `stack_mut`) `pub(crate)`.

I assume that the main reason why `stack` was originally public was to give the machine access to the stack so that this diagnostics function can be implemented. Since now the stack is in the machine, is there still a need to have `InterpCx::stack` public?

Reply:
Oh lol, I also entirely missed than when reviewing the rsutc PR.^^

I supposed the getter in `InterpCx` is no longer needed now, you are right. This also means that we can be sure that we consistently use `active_thread_stack` and not `stack`, so there's no weird "two ways to do things", so... let's leave this here as-is.